ios/RCTWebRTC/SocketConnection.m

```diff
@@ -0,0 +1,173 @@
+//
+//  SocketConnection.m
+//  RCTWebRTC
+//
+//  Created by Alex-Dan Bumbu on 08/01/2021.
+//
+
+#include <sys/socket.h>
+#include <sys/un.h>
+
+#import "SocketConnection.h"
+
+const NSUInteger kConnectionBufferSize = 10*1024;
+
+@interface SocketConnection ()
+
+@property (nonatomic, assign) int serverSocket;
+@property (nonatomic, strong) dispatch_source_t listeningSource;
+
+@property (nonatomic, strong) NSThread *networkThread;
+
+@property (nonatomic, strong) NSInputStream *inputStream;
+@property (nonatomic, strong) NSOutputStream *outputStream;
+
+@end
+
+@implementation SocketConnection
+
+- (instancetype)initWithFilePath:(nonnull NSString *)filePath {
+    self = [super init];
+    
+    [self setupNetworkThread];
+        
+    self.serverSocket = socket(AF_UNIX, SOCK_STREAM, 0);
+    if (self.serverSocket < 0) {
+       NSLog(@"failure creating socket");
+       return nil;
+    }
+    
+    if (![self setupSocketBufferSize]) {
+        return  nil;
+    }
+    
+    if (![[NSFileManager defaultManager] createFileAtPath:filePath contents:nil attributes:nil]) {
```

Comment:
Why are you creating the file just to unlink it later, before you bind?

Reply:
fixed